I found that some combinations of grad_method and qfi_method do not pass the unit test. Could you take a look at it? @Zoufalc @BryceFuller

E.g.,
```
======================================================================
FAIL: test_natural_gradient_10___lin_comb____overlap_block_diag____perturb_diag_elements__ (test.aqua.operators.test_gradients.TestGradients)
Test the natural gradient
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/ima/envs/qopt/lib/python3.7/site-packages/ddt.py", line 182, in wrapper
    return func(self, *args, **kwargs)
  File "/Users/ima/tasks/0_2020/qiskit/aqua/test/aqua/operators/test_gradients.py", line 398, in test_natural_gradient
    decimal=0)
  File "/Users/ima/envs/qopt/lib/python3.7/site-packages/numpy/testing/_private/utils.py", line 1044, in assert_array_almost_equal
    precision=decimal)
  File "/Users/ima/envs/qopt/lib/python3.7/site-packages/numpy/testing/_private/utils.py", line 840, in assert_array_compare
    raise AssertionError(msg)
AssertionError: 
Arrays are not almost equal to 0 decimals

Mismatched elements: 1 / 2 (50%)
Max absolute difference: 3.13933993
Max relative difference: 0.74746189
 x: array([-1.e+00,  1.e-07])
 y: array([-4.,  0.])
```

--- a/test/aqua/operators/test_gradients.py
+++ b/test/aqua/operators/test_gradients.py
@@ -369,10 +369,9 @@ class TestGradients(QiskitAquaTestCase):
                                                  correct_values[i], decimal=1)
 
     @idata(product(['lin_comb', 'param_shift', 'fin_diff'],
-                   ['lin_comb_full', 'overlap_block_diag', 'overlap_diag'],
                    [None, 'lasso', 'ridge', 'perturb_diag', 'perturb_diag_elements']))
     @unpack
-    def test_natural_gradient(self, grad_method, qfi_method, regularization):
+    def test_natural_gradient(self, grad_method, regularization):
         """Test the natural gradient"""
         ham = 0.5 * X - 1 * Z
         a = Parameter('a')
@@ -387,7 +386,6 @@ class TestGradients(QiskitAquaTestCase):
 
         op = ~StateFn(ham) @ CircuitStateFn(primitive=qc, coeff=1.)
         nat_grad = NaturalGradient(grad_method=grad_method,
-                                   qfi_method=qfi_method,
                                    regularization=regularization).convert(operator=op,
                                                                           params=params)
         values_dict = [{params[0]: np.pi / 4, params[1]: np.pi / 2}]

--- a/test/aqua/operators/test_gradients.py
+++ b/test/aqua/operators/test_gradients.py
@@ -411,7 +411,7 @@ class TestGradients(QiskitAquaTestCase):
     @idata(product(['lin_comb', 'param_shift', 'fin_diff'],
                    [None, 'lasso', 'ridge', 'perturb_diag', 'perturb_diag_elements']))
     @unpack
-    def test_natural_gradient(self, grad_method, regularization):
+    def test_natural_gradient(self, method, regularization):
         """Test the natural gradient"""
         ham = 0.5 * X - 1 * Z
         a = Parameter('a')
@@ -425,7 +425,7 @@ class TestGradients(QiskitAquaTestCase):
         qc.rx(params[1], q[0])
 
         op = ~StateFn(ham) @ CircuitStateFn(primitive=qc, coeff=1.)
-        nat_grad = NaturalGradient(grad_method=grad_method,
+        nat_grad = NaturalGradient(grad_method=method,
                                    regularization=regularization).convert(operator=op,
                                                                           params=params)
         values_dict = [{params[0]: np.pi / 4, params[1]: np.pi / 2}]

